### ISSUE TITLE
Make query strategy conform to sklearn guidelines for estimators

### ISSUE THREAD

**Issue by tpham93** [scikit-activeml/scikit-activeml#42]
see here: https://scikit-learn.org/stable/developers/develop.html
missing things are:
- move validation from __init__(...) to query(...)
- default arguments for all parameters of __init__(...)

**Comment by mherde** [comment]
Randomstate shall be set only once.

**Comment by tpham93** [comment]
pool.RandomSampling should generate a new random state in each query call and make the seed dependent on the length of candidates, so that query calls for evaluation should not influence the result.
